Source/USemLog/Private/Control/SLControlManager.cpp

```diff
@@ -72,20 +82,62 @@ void ASLControlManager::StartSimulationSelectionOnly(const TArray<FString>& Ids)
 				*FString(__FUNCTION__), __LINE__, *GetName());
 			continue;
 		}
-		//if (!Individual->IsMovable())
-		//{
-		//	UE_LOG(LogTemp, Log, TEXT("%s::%d Individual (%s) is not movable.."),
-		//		*FString(__FUNCTION__), __LINE__, *Id);
-		//	return;
-		//}
+
 		AActor* Actor = Individual->GetParentActor();
 		UStaticMeshComponent* StaticMesh = Cast<UStaticMeshComponent>(Actor->GetComponentByClass(UStaticMeshComponent::StaticClass()));
 		StaticMesh->SetSimulatePhysics(true);
 	}
+
+	bIsSimStart.AtomicSet(true);
+	OnSimulationStart.ExecuteIfBound();
+
+	if (Seconds > 0)
+	{
+		StopSimulationSelectionOnly(Ids, Seconds);
+	}
+	
+	return true;
 }
 
-void ASLControlManager::StopSimulationSelectionOnly(const TArray<FString>& Ids)
+// Stop physics simulation on individuals with delay
+bool ASLControlManager::StopSimulationSelectionOnly(const TArray<FString>& Ids, int32 Seconds)
 {
+	if (!IsSimStart())
+		return false;
+
+	if (Seconds > 0)
+	{
+		FTimerHandle TimerHandle;
+		FTimerDelegate TimerDelegateDelay;
+		TimerDelegateDelay.BindLambda([this](TArray<FString> Ids) {
+			StopSimulationSelectionOnly(Ids);
+		}, Ids);
+		GetWorld()->GetTimerManager().SetTimer(TimerHandle, TimerDelegateDelay, Seconds, false);
+		OnSimulationStopCountDown.ExecuteIfBound(Seconds);
+	}
+	else
+	{
+		if (!bIsSimStart.AtomicSet(false))
+		{
+			StopSimulationSelectionOnly(Ids);
+		}
+	}
+
+	return true;
+}
+
+// Stop physics simulation on individuals without delay
+bool ASLControlManager::StopSimulationSelectionOnly(const TArray<FString>& Ids)
+{
+	UE_LOG(LogTemp, Warning, TEXT("%s::%d Individual ) sssssssssssd111111.."),
```

Comment:
type probably, not sure if it will compile because of it

Reply:
this is for testing. I will remove them